### DISCUSSION TITLE
WalletConnect V2 - Trust wallet cannot parse the transaction received from the Dapp

### DISCUSSION THREAD

**Discussion by ineskh** [trustwallet/wallet-core#3209] [Q&A]
**Describe the bug**
Cannot receive the transaction sent from a Dapp on my Trust wallet (iOS version) and there is no error in return. 
It's like the Trust Wallet is not able to parse the transaction like tx  = { from, to,  nonce, gas ,  value ,  chainId}  

**To Reproduce**
1- Install walletConnect V2 on a Dapp using "@walletconnect/ethereum-provider"
2- Connect the Trust Wallet (iOS) to the Dapp by  scanning the QRcode (Fantom network for example)  : Connection is Ok
3- Send the transaction  tx  = { from, to,  nonce, gas,  value,  chainId}  to the wallet using : 
web3.eth
        .sendTransaction(tx)
        .once("transactionHash", (txHash: string) => resolve(txHash))
        .on('receipt', async function(receipt:any){
        })
        .catch((err: any) => {console.error(err); reject(err)});

==> TrustWallet does  not receive the transaction 
However when I use the walletConnect V1, the wallet connection and the transaction confirmation are Ok

**Expected behavior**
I expect that the trust wallet receives the transaction for confirmation 

**Screenshots**

**Additional context**

**Comment by ineskh** [comment]
I fixed this problem by adding the parameter data : "0x0" to the transaction when it is a transfer of coin(eth, bnb, ftm...) and not token (erc20/beb20)

**Comment by mubshrx** [comment]
Yeah, we faced the same issue and took us one day to figure out that we need to put data: "0x" in order to send network's native currency.
